### DISCUSSION TITLE
Help a newbie: How to execute an executable?

### DISCUSSION THREAD

**Discussion by mathomp4** [community/community#25537] [Actions]
<p>All,</p>
<p>I feel really stupid right now, but for the life of me I can’t figure out how to do the most <em>basic</em> things with GitHub Actions. To play around, I setup a very boring Fortran project (<a href="https://github.com/mathomp4/PreciseSum" rel="nofollow noopener">https://github.com/mathomp4/PreciseSum</a>) and in there I have a code that you can do ‘make’ and the run ‘./test.exe’.</p>
<p>So, I tried to follow a few examples online to build it (no Fortran in the <a href="https://github.com/actions/starter-workflows/tree/master/ci" rel="nofollow noopener">starter workflows</a>), and got to:</p>
<pre><code>name: CI

on: [push]

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
    - uses: actions/checkout@v1
    - name: Install gfortran
      run: |
         sudo add-apt-repository ppa:ubuntu-toolchain-r/test
         sudo apt update
         sudo apt-get install gfortran-9
    - name: Show gfortran version
      run: gfortran-9 --version
    - name: Build
      run: make FC=gfortran-9
    - name: List files
      run: ls
    - name: Run
      run: ./text.exe
</code></pre>
<p>It seems to build just fine, but trying to execute it fails:</p>
<pre><code>./text.exe
  shell: /bin/bash -e {0}
/home/runner/work/_temp/33162838-038b-4046-83df-f2df171cf5a1.sh: line 1: ./text.exe: No such file or directory
##[error]Process completed with exit code 127.
</code></pre>
<p>How does one do this with GitHub Actions? Or can one?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>You said that “I have a code that you can do ‘make’ and the run ‘./test.exe’. “</p>
<p>After build the Fortran project, test.exe is generated . And you can see it in List files step.</p>
<p>In run step, you use wrong <strong>text.exe.</strong> Correct it to <strong>test.exe</strong> will fix your issue.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c881dfc7b914f4e45dfaefa10ca1c9f7149468dd.png" data-download-href="/uploads/short-url/sBLKnU5ZL2KL7eXeuVeep9CH02V.png?dl=1" title="runtestexe.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181091749-06c44472-7dcf-45c9-a9e2-e3cfdb35cafa.png" alt="runtestexe.png" data-base62-sha1="sBLKnU5ZL2KL7eXeuVeep9CH02V" width="473" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c881dfc7b914f4e45dfaefa10ca1c9f7149468dd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">runtestexe.png</span><span class="informations">632×668 12 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by mathomp4** [comment]
<p>Sigh. I am an idiot. Looking at all the hard issues and searching docs and, typo!</p>
